WebKit does not forward debug/info logging from the WebContent process

cdumez opened the pull request:
#### b5ffe05c57be2895f42ae7da6f6756fd4b971bb1
<pre>
WebKit does not forward debug/info logging from the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=298087">https://bugs.webkit.org/show_bug.cgi?id=298087</a>
<a href="https://rdar.apple.com/152591955">rdar://152591955</a>

Reviewed by Per Arne Vollan.

We used to only forward debug/info logging from the WebContent processes in
debug builds. While this may be OK for WebKit developers, this is insufficient
for underlying framework developers as they would generally use a release build
of WebKit.

To address the issue, we now rely on `os_log_debug_enabled(OS_LOG_DEFAULT)`
instead to determine if we should forward debug/info messages or not. This is
true when the developer is streaming debug logs or debug logging has been
explicitly enabled system wide (not the default).

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::registerLogClient):
(WebKit::WebProcess::initializeLogForwarding):

Canonical link: <a href="https://commits.webkit.org/299379@main">https://commits.webkit.org/299379@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7d2b30ba77ec98247a23ddb8d2416c4b046eefa5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118599 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38280 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28931 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124776 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70658 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/50e73696-96fd-49bb-b2ef-13a7f55d6833) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38976 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46862 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90014 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59561 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cf4fd8aa-6afb-4eca-b932-f8188a0b83b8) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121552 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31045 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106329 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70518 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bdbf0996-96c3-4a5d-bcfa-a237659c40a4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30102 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68433 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100483 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24631 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127837 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45506 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34333 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98647 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45870 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102550 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98431 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25064 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43872 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21867 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42004 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45376 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51054 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44839 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48186 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46526 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f377efc0edd6c6322956347dd8b74aab5d394457. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50054

cdumez committed:
WebKit does not forward debug/info logging from the WebContent process
https://bugs.webkit.org/show_bug.cgi?id=298087
rdar://152591955

Reviewed by Per Arne Vollan.

We used to only forward debug/info logging from the WebContent processes in
debug builds. While this may be OK for WebKit developers, this is insufficient
for underlying framework developers as they would generally use a release build
of WebKit.

To address the issue, we now rely on `os_log_debug_enabled(OS_LOG_DEFAULT)`
instead to determine if we should forward debug/info messages or not. This is
true when the developer is streaming debug logs or debug logging has been
explicitly enabled system wide (not the default).

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::registerLogClient):
(WebKit::WebProcess::initializeLogForwarding):

Canonical link: https://commits.webkit.org/299379@main
